codebreaker_garage/lib/codebreaker_garage/console.rb

```diff
@@ -29,16 +30,20 @@ def want_to_save
       replay
     end
 
+    def correct_input
+      user_guessing = []
+      @user_tries_to_guess.each_char {|char| user_guessing.push char.to_i }
+      result = @game.give_result(user_guessing)
+      puts result
+      want_to_save if result == "++++"
```

Comment:
No, it saves in both cases. When user wins(guesses the code) and when user loses(no more attemts left).